### DISCUSSION TITLE
How to generate header files from the XML file for boards other than Icicle Kit?

### DISCUSSION THREAD

**Discussion by ManishaI31214** [polarfire-soc/polarfire-soc-documentation#116] [Bare metal embedded software] [answered]
We are trying to generate the header files from the XML file for DDR4 using the MPFS HAL DDR demo given at the below link:
https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples/tree/main/driver-examples/mss/mpfs-hal/mpfs-hal-ddr-demo

We have updated the XML file for “ddr-hs-base-board-ddr4” in the boards directory.

However, when we build the SoftConsole project, the header files are not getting generated for the respective XML file.

For Icicle Kit, we are able to successfully generate the header files from XML file.

Can you kindly let us know how to generate the header files from the XML file for other boards.

**Comment by hughbreslin** [comment, marked as answer]
Hi @ManishaI31214 

For the debug configuration:

It looks like the build step to generate the header files from the XML is missing from this build configuration. To fix this you need to update the build steps for the configuration to generate the header files:

![image](https://user-images.githubusercontent.com/44862881/177546884-71b412be-fc7c-4077-83c4-9eae079fc82d.png)

The full command I used is: `${env_var:MACRO_PYTHON_BINARY_PATH_AND_EXECUTABLE} ../src/platform/soc_config_generator/mpfs_configuration_generator.py ../src/boards/ddr-hs-base-board-ddr4/fpga_design/xml   ../src/boards/ddr-hs-base-board-ddr4`

For the release configuration I can see it points to the Icicle Kit XML so if you're building for release you'll need to update the release configuration with the command above
